lib/ngx/balancer.lua

```diff
@@ -15,22 +16,66 @@ local getfenv = getfenv
 local error = error
 local type = type
 local tonumber = tonumber
+local max = math.max
+local subsystem = ngx.config.subsystem
+local ngx_subsystem_lua_ffi_balancer_set_current_peer
+local ngx_subsystem_lua_ffi_balancer_set_more_tries
+local ngx_subsystem_lua_ffi_balancer_get_last_failure
+
+
+if subsystem == 'http' then
+    ffi.cdef[[
+    int ngx_http_lua_ffi_balancer_set_current_peer(ngx_http_request_t *r,
+        const unsigned char *addr, size_t addr_len, int port, char **err);
+
+    int ngx_http_lua_ffi_balancer_set_more_tries(ngx_http_request_t *r,
+        int count, char **err);
+
+    int ngx_http_lua_ffi_balancer_get_last_failure(ngx_http_request_t *r,
+        int *status, char **err);
+
+    int ngx_http_lua_ffi_balancer_set_timeouts(ngx_http_request_t *r,
+        long connect_timeout, long send_timeout,
+        long read_timeout, char **err);
+    ]]
+
+    ngx_subsystem_lua_ffi_balancer_set_current_peer =
```

Comment:
Shall we use the prefix `ngx_lua_ffi_` instead of `ngx_subsystem_lua_ffi_`? The current prefix is just too long :)

Please change the other similar places.